packages/lib/data/ModInfo.js

```diff
@@ -0,0 +1,231 @@
+"use strict";
+const fs = require("fs-extra");
+const JSZip = require("jszip");
+const path = require("path");
+
+const libHash = require("../hash");
+const libSchema = require("../schema");
+const { findRoot } = require("../zip_ops");
+
+const { integerModVersion, integerFactorioVersion, modVersionRegExp } = require("./version");
+
+
+/**
+ * Info about a mod available on the master server.
+ *
+ * See https://wiki.factorio.com/Tutorial:Mod_structure#info.json for
+ * details of the invidiual fields sourced from info.json.
+ * @alias module:lib/data.ModInfo
+ */
+class ModInfo {
+	/**
+	 * Internal name of this mod.
+	 * This is the name of the zip file as well as the name that appears in
+	 * info.json.
+	 * @type {string}
+	 */
+	name = "";
+
+	/**
+	 * Version of the mod.
+	 * Sourced from info.json.
+	 * @type {string}
+	 */
+	version = "";
+
+	/**
+	 * Integer representation of the version
+	 * @type {number}
+	 */
+	get integerVersion() {
+		return integerModVersion(this.version);
+	}
+
+	/**
+	 * Display name of the mod.
+	 * Sourced from info.json.
+	 * @type {string}
+	 */
+	title = "";
+
+	/**
+	 * Author of the mod.
+	 * Sourced from info.json.
+	 * @type {string}
+	 */
+	author = "";
+
+	/**
+	 * Contact field for mod.
+	 * Sourced from info.json.
+	 * @type {string}
+	 */
+	contact = "";
+
+	/**
+	 * Homepage of the mod.
+	 * Sourced from info.json.
+	 * @type {string}
+	 */
+	homepage = "";
+
+	/**
+	 * Description of the mod.
+	 * Sourced from info.json.
+	 * @type {string}
+	 */
+	description = "";
+
+	/**
+	 * Major version of Factorio this mod supports.
+	 * Sourced from info.json.
+	 * @type {string}
+	 */
+	factorioVersion = "0.12";
```

Comment:
Why 0.12? We are far far far away from the time 0.12 was last relevant :D

Reply:
It's what the game interprets a mod without `factorio_version` as being for.